basyx/aas/adapter/http.py

```diff
@@ -100,49 +100,63 @@ class StrippedResultToJsonEncoder(ResultToJsonEncoder):
 
 class APIResponse(abc.ABC, Response):
     @abc.abstractmethod
-    def __init__(self, obj: Optional[ResponseData] = None, stripped: bool = False, *args, **kwargs):
+    def __init__(self, obj: Optional[ResponseData] = None, cursor: Optional[int] = None,
+                 stripped: bool = False, *args, **kwargs):
         super().__init__(*args, **kwargs)
         if obj is None:
             self.status_code = 204
         else:
-            self.data = self.serialize(obj, stripped)
+            self.data = self.serialize(obj, cursor, stripped)
 
     @abc.abstractmethod
-    def serialize(self, obj: ResponseData, stripped: bool) -> str:
+    def serialize(self, obj: ResponseData, cursor: int, stripped: bool) -> str:
         pass
 
 
 class JsonResponse(APIResponse):
     def __init__(self, *args, content_type="application/json", **kwargs):
         super().__init__(*args, **kwargs, content_type=content_type)
 
-    def serialize(self, obj: ResponseData, stripped: bool) -> str:
-        return json.dumps(obj, cls=StrippedResultToJsonEncoder if stripped else ResultToJsonEncoder,
+    def serialize(self, obj: ResponseData, cursor: int, stripped: bool) -> str:
+        data: Dict[str, Any] = {}
+
+        # Add paging metadata if cursor is not None
+        if cursor is not None:
+            data["paging_metadata"] = {"cursor": cursor}
+
+        # Add the result
+        data["result"] = obj
```

Comment:
If we always add a wrapping object, we aren't conforming to the specification anymore. Instead, the wrapping object should only be added in case multiple objects are serialized. Although XML responses aren't standardized, the same should be applied there so we're at least consistent in this regard.